org/postgresql/util/PGTimestamp.java

```diff
@@ -0,0 +1,134 @@
+/*-------------------------------------------------------------------------
+ *
+ * Copyright (c) 2004-2014, PostgreSQL Global Development Group
+ *
+ *
+ *-------------------------------------------------------------------------
+ */
+package org.postgresql.util;
+
+import java.sql.PreparedStatement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.Calendar;
+import java.util.TimeZone;
+
+/**
+ * This class augments the Java built-in Timestamp to allow for explicit setting
+ * of the time zone.
+ */
+public class PGTimestamp extends Timestamp
+{
+    /** The serial version UID. */
+    private static final long serialVersionUID = -6245623465210738466L;
+
+    /** The optional calendar for this timestamp. */
+    private Calendar calendar;
+
+    /**
+     * Constructs a <code>PGTimestamp</code> without a time zone. The integral
+     * seconds are stored in the underlying date value; the fractional seconds
+     * are stored in the <code>nanos</code> field of the <code>Timestamp</code>
+     * object.
+     *
+     * @param time
+     *            milliseconds since January 1, 1970, 00:00:00 GMT. A negative
+     *            number is the number of milliseconds before January 1, 1970,
+     *            00:00:00 GMT.
+     * @see Timestamp#Timestamp(long)
+     */
+    public PGTimestamp(long time)
+    {
+        this(time, null);
+    }
+
+    /**
+     * Constructs a <code>PGTimestamp</code> with the given time zone. The
+     * integral seconds are stored in the underlying date value; the fractional
+     * seconds are stored in the <code>nanos</code> field of the
+     * <code>Timestamp</code> object.
+     * <p>
+     * The calendar object is optional. If absent, the driver will treat the
+     * timestamp as <code>timestamp without time zone</code>. When present, the
+     * driver will treat the timestamp as a <code>timestamp with time zone</code>
+     * using the <code>TimeZone</code> in the calendar object. Furthermore, this
+     * calendar will be used instead of the calendar object passed to
+     * {@link PreparedStatement#setTimestamp(int, Timestamp, Calendar)}.
+     *
+     * @param time
+     *           milliseconds since January 1, 1970, 00:00:00 GMT. A negative
+     *           number is the number of milliseconds before January 1, 1970,
+     *           00:00:00 GMT.
+     * @param calendar
+     *           the calendar object containing the time zone or
+     *           <code>null</code>.
+     * @see Timestamp#Timestamp(long)
+     */
+    public PGTimestamp(long time, Calendar calendar)
+    {
+        super(time);
+        this.setCalendar(calendar);

```

Comment:
Calling non-`final` methods in constructor is a code smell (the object is not yet initialized, thus impact of overridden `setCalendar` is not predictable). Can you please make `setCalendar` final just in case?